### PR TITLE
Prepare for 1.0 release

### PR DESCRIPTION
**Pull request by cecton**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -92,6 +92,7 @@ const MAX_ALIGN: u64 = 16384;
 
 /// An error that can be produced while reading, writing or managing a GPT.
 #[derive(Debug, Error)]
+#[non_exhaustive]
```
Comment (cecton) [inline src/lib.rs:95]: I realized the other day I made a patch release after the addition of a feature (#76) but I failed to see that adding variants to an enum is actually a breaking change in the API.

I think those error structs should be non-exhaustive so the API of that particular object can change over time.

Do you think that's correct/better @bgilbert?

Comment (bgilbert) [reply src/lib.rs:95]: Yes, I agree. :+1:

**Comment by cecton** [comment]
@bgilbert I think you contributed a lot to this project and I probably asked you already but, would like to become an owner on this project? I would like to reduce the bus factor, you won't have to do much maintenance on it. I barely touch it myself.

If you agree, let me know your user on crates.io and I will send an invitation. I will also probably move it to its own organization. There is also the project mbrman where I am the single owner and would gladly accept a peer if you want.

**Comment by bgilbert** [comment]
I declined last time you asked me, but yeah, I agree that reducing the bus factor would be good. :+1: I'm bgilbert over on crates.io, and I'm okay with being an owner for both gptman and mbrman.  Moving them to an org seems ideal, but I'm okay either way.

**Comment by bgilbert** [comment]
FYI, I've received the invitations to the crates.io crates and to the GitHub org, but it looks like I don't have any new permissions to the GitHub repos.

**Comment by cecton** [comment]
> FYI, I've received the invitations to the crates.io crates and to the GitHub org, but it looks like I don't have any new permissions to the GitHub repos.

:heavy_check_mark: fixed. Can you check again?

**Comment by bgilbert** [comment]
Looks good!  :+1:

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,23 @@
...
+- cargo: Widen nix dependency range (#91)
+- Add monthly workflow to update dependencies (#89)
+- Update dependencies (#88)
+- Bump thiserror from 1.0.30 to 1.0.31 (#85)
+- Bump rand from 0.8.4 to 0.8.5 (#84)
+- Small infrastructure improvements (#82)
+- [CLI] Replace structopt with clap 3.1 (#77)
+- Consolidate workflow jobs; test library with 1.46 MSRV and CLI with 1.54 (#79)
+- [BREAKING] cargo: Disable CLI by default (#81)
+- cargo: Update crc to 3.0.0 (#80)
```
Comment (cecton) [inline CHANGELOG.md:19]: @bgilbert can you check this? I just copy-pasted the commits but I added CLI and BREAKING to flag things a bit

**Comment by cecton** [comment]
(Hopefully that will be the last time I have a branch on this repo. Next branches will use my fork)

File: CHANGELOG.md (hunk 2 above)
Comment (bgilbert) [reply CHANGELOG.md:19]: How do you normally prefer to handle changelogs?  I usually try to trim down/clean up the commit history to make it more readable for users, but I've worked on projects that just list the commits and that's okay too.

Comment (cecton) [reply CHANGELOG.md:19]: I think (and I hope) there will be almost no more commit in the future. So I expect the list to be limited. This one release is the last one with a lot of changes. So I guess we could simply go with listing all the commits.

I did take the habit to add a context for the commit so people see what they need to be aware of when updating.... :thinking: it's true that a lot of those adds noise and make things less readable...

Make I can remove those related to CI or dependency updates?

Comment (cecton) [reply CHANGELOG.md]: Pushed in e7fd4cb

**Comment by cecton** [comment]
@bgilbert is there anything you can think of for the 1.0?

Otherwise I will merge this and release using cargo-release

File: CHANGELOG.md (hunk 2 above)
Comment (bgilbert) [reply CHANGELOG.md]: Yeah, I agree with dropping the CI updates and at least consolidating the dependency ones.  Here's what I came up with:

```
- [BREAKING] Disable `cli` feature by default (#81)
- [BREAKING] Make `Error` type `non_exhaustive` so the API can be stabilized better (#94)
- [BREAKING CLI]: Have `--output` take lowercase field names as arguments (#77)
- [CLI] Replace structopt with clap 3.1 (#77)
- Define library MSRV of 1.46 and CLI MSRV of 1.54 (#79)
- Remove useless allocation (#93)
- Update repository URL (#82)
- Update dependencies
```

Feel free to use/ignore/edit as you'd prefer.

**Comment by bgilbert** [comment]
See changelog proposal above. ^  Other than that, all good to release AFAIK. :+1:

File: CHANGELOG.md (hunk 2 above)
Comment (cecton) [reply CHANGELOG.md]: Ok I'll use that. Seems better

**Review by bgilbert — APPROVED** [review]

**Comment by bgilbert** [comment]
Congratulations on reaching 1.0!  When [coreos-installer](https://github.com/coreos/coreos-installer) needed to parse (and later rewrite) GPTs, gptman was very easy to integrate, without any of the strange behavior or missing features that sometimes exist in relatively niche crates.  We're having gptman do some unusual partition table rewriting that could cause customer data loss if something goes wrong, and it's been completely reliable and easy to work with.  Thanks for making the library, and for being so incredibly responsive to PRs and feedback!

**Comment by cecton** [comment]
Thanks a lot!! :heart:

**Pull request merged**